<?xml version="1.0"?>
<launch>

    <arg name="dev" default="/dev/ttyACM0"/>
    <arg name="baud" default="38400"/>
    <arg name="address" default="128"/>
    <arg name="max_speed" default="1.0"/>
    <arg name="ticks_per_meter" default="2495"/>
    <arg name="base_width" default="0.421"/>
    <arg name="run_diag" default="true"/>
    <arg name="invert_motor_direction" default="false"/>
    <arg name="flip_left_and_right_motors" default="false"/>

    <node if="$(arg run_diag)" pkg="roboclaw_node" type="roboclaw_node.py" name="roboclaw_node">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address" value="$(arg address)"/>
        <param name="~max_speed" value="$(arg max_speed)"/>
        <param name="~ticks_per_meter" value="$(arg ticks_per_meter)"/>
        <param name="~base_width" value="$(arg base_width)"/>
        <param name="~invert_motor_direction" value="$(arg invert_motor_direction)"/>
        <param name="~flip_left_and_right_motors" value="$(arg flip_left_and_right_motors)"/>
    </node>

    <node pkg="diagnostic_aggregator" type="aggregator_node"
          name="diagnostic_aggregator">
        <rosparam command="load"
                  file="$(find roboclaw_node)/config/roboclaw_diag.yaml"/>
    </node>

</launch>
